src/main/java/org/wildfly/security/sasl/digest/DigestClientFactory.java

```diff
@@ -60,6 +62,9 @@ public DigestClientFactory(final Provider provider) {
     public SaslClient createSaslClient(String[] mechanisms, String authorizationId, String protocol, String serverName,
             Map<String, ?> props, CallbackHandler cbh) throws SaslException {
 
+        if (cbh == null) throw ElytronMessages.log.mechCallbackHandlerMissing().toSaslException();
```

Comment:
Some of the JDK implementations also throw NPE if a parameter is null; essentially it's pretty arbitrary and not a part of the API contract defined by either Oracle, and we define our method call contract to throw IAE with a standard i18n exception message and code when a parameter is null due to a programming error.  This makes debugging easier and also helps manage the user's expectations.